### PR TITLE
Remove apparent contradiction in vector.h header

### PR DESCRIPTION
**Pull request by esainane**
3205a92ad8 was a major commit which removed `PoolVector`, and replaced most references to `PoolVector` with `Vector` instead. In most cases, this was appropriate, given that `PoolVector` was being replaced with `Vector`, as an effective generalist in 64-bit address space layouts.

However, vector.h itself was left with an artifact advising the reader to use `Vector` instead of `Vector` for large arrays. While this led to a fascinating deep dive, and hopefully improved some of the documentation along the way, it's probably best to clean this up for the next person.

### CODE REVIEW

File: core/templates/vector.h [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 /**
  * @class Vector
- * Vector container. Regular Vector Container. Use with care and for smaller arrays when possible. Use Vector for large arrays.
+ * Vector container. Simple, copy-on-write, array backed container.
```
Comment (AThousandShips) [inline core/templates/vector.h:36]: It isn't "array backed", this should probably suggest using `LocalVector` when appropriate instead

Comment (esainane) [reply core/templates/vector.h]: Will do!

**Review by Chubercik — APPROVED** [review]

**Review by fire — APPROVED** [review]

**Review by clayjohn — APPROVED** [review]

**Comment by clayjohn** [comment]
Looks great! The final step before merging is to squash all the commits together so that the whole PR only contains 1 big commit with all your changes. We like to merge one commit at a time to keep the git history clean and navigable. 

If you don't know how to do that, we have a helpful tutorial in the official documentation https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

**Comment by esainane** [comment]
Done!

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**